Remove legacy css (closes #126)

housseindjirdeh opened the pull request:
Most of the styles are global and so I just migrated to aphrodite-shortcomings.css. There were a couple that were only referenced once so I migrated those in-line to their respective components:

```
.container {
   margin-top: 40px;
   margin-left: auto;
   margin-right: auto;
   margin-bottom: 100px;
   max-width: 1000px;
   width: 90%;
}
```

^ These styles were only referenced once (due to specificity) in the episodes page and so I just migrated them to `EpisodePage`

Also, both `+space-children*` and `hr.person-note-separator` were only being referenced once so I moved each of them to their components with aphrodite.

Commit:
Remove unused styles in legacy css. For #126

Commit:
Merge branch 'master' into remove-legacy-css

Commit:
Added space children class back to episodes page

Commit:
Migrated to components in episode pages

Commit:
Removed legacy css file, finished migrating singleton styles

Commit:
Fixed merge conflicts

Commit:
Final commit, closes #126

kentcdodds commented on resources/css/aphrodite-shortcomings.css:
```diff
@@ -6,12 +6,79 @@ body {
   }
 }
+
+.container {
```
Do these styles have to be nested in `.container`? I'd love to avoid nesting as much as possible if we can.

kentcdodds replied:
Mostly, I'd just love to get rid of references to `container` in the components. I'd prefer to not have to have any `className`s in the components that are not aphrodite class names :-)

kentcdodds commented on src/pages/episode/sections/show-notes/index.js:
```diff
@@ -40,3 +43,17 @@ function PeopleNotes({people = []}) {
+PeopleNotes.styles = StyleSheet.create({
+  noteSeparator: {
+    width: '90%',
+    '@media only screen and (max-width: 767px)': {
```
Any way we could keep these media queries in the `media-queries` file? I'd prefer to have them all in the same place and give them a descriptive name. If it's close to an existing one, let's try to reuse existing media queries as well.

kentcdodds commented on src/pages/episode/sections/show-notes/person-notes.js:
```diff
@@ -48,7 +48,17 @@ PersonNotes.styles = StyleSheet.create({
-  notesContainer: {flex: 1},
+  notesContainer: {
+    flex: 1,
+    '@media only screen and (max-width: 767px)': {
```
Same here :+1:

kentcdodds commented:
This is looking great! Thanks for working on this!

A bit of feedback. You might notice that your pull request has several unrelated commits in it:

<img width="711" alt="screen shot 2016-06-22 at 6 54 44 pm" src="https://cloud.githubusercontent.com/assets/1500684/16287699/cf2bd640-38aa-11e6-915a-3dd396a14303.png">

When I merge your PRs, I normally just use GitHub's [squash and merge](https://help.github.com/articles/about-pull-request-merge-squashing/) feature, so these ultimately end up as a single commit, so it's not really a big deal. But by [squashing your commits](https://github.com/blog/2141-squash-your-commits) and rebasing them onto master before you push, it makes it a bit easier for you to update the PR until it's merged.

I think you've mentioned that you've watched my egghead course, but just in case, here are some relevant lessons that could help your workflow:
- [Fork, clone, set upstream](https://egghead.io/lessons/javascript-how-to-fork-and-clone-a-github-repository?course=how-to-contribute-to-an-open-source-project-on-github)
- [rebase](https://egghead.io/lessons/javascript-how-to-rebase-a-git-pull-request-branch?course=how-to-contribute-to-an-open-source-project-on-github)
- [squash](https://egghead.io/lessons/javascript-how-to-squash-multiple-git-commits?course=how-to-contribute-to-an-open-source-project-on-github)

Again, it makes no difference ultimately, I just thought it could help your workflow :-) Thanks again!

kentcdodds commented:
Oh, and [this](https://egghead.io/lessons/misc-git-how-to-change-the-most-recent-commit-with-amend) also might be helpful to you (shows you how to change the commit message, but can also be used to add changes to the most recent commit as well)

housseindjirdeh commented:
@kentcdodds Genuinely can't thank you enough :)

Yeah was going to ask you about squashing commits actually, thanks again for the links! I'll make sure to clean up the commits for this PR and all the next ones (and I never knew about `amend`, I've always been so used to adding commits on top of each-other if I've forgotten something or made a mistake to that's going to help a ton:O).

Will update this today/tomorrow and move the media-queries to their appropriate file and see if I can remove the nested `.container.` Thanks again!

Commit:
Removed nested container and margin-bottom-large class

Commit:
Merge branch 'master' into remove-legacy-css

housseindjirdeh commented:
1) Removed `.container` nesting. The homepage is essentially the only page that wasn't nested with this class so had to do some small modifications to other styling files to make sure there were no changes to the overall style.

Note: The only difference now is in the sponsors section which I gave a little spacing (looks a little better now I think).

## Sponsors (previous)

![sponsors previous](https://cloud.githubusercontent.com/assets/12476932/16317609/e8f52040-3959-11e6-8cc3-a254c0e148ae.png)

## Sponsors(current)

![sponsors current](https://cloud.githubusercontent.com/assets/12476932/16317623/f0cfcc70-3959-11e6-9fb2-ec122021631d.png)

2) Moved media query calls to the media queries file (created two new instances, `upToMediumBig` and `atLeastMediumBig`

3) Noticed that `+margin-bottom-large` was used in a few instances in the site so don't think it needs to be in a global css file. Removed it and created aphrodite `classnames` for each of the components 

Thanks again for the help! I'll make sure my next PRs have better workflow commits before I push to make things a little cleaner :). Let me know if need me to change anything here though :+1:

kentcdodds commented on resources/css/aphrodite-shortcomings.css:
```diff
@@ -6,12 +6,70 @@ body {
   }
 }
+
+h1, h2, h3, h4 {
```
This looks so much cleaner! Thank you!

kentcdodds commented on resources/css/styles.css:
```diff
@@ -1,4 +1,3 @@
 @import 'reset.css';
-@import 'legacy.css';
```
:tada:

kentcdodds commented on src/components/sponsors/sponsor-group.js:
```diff
@@ -39,4 +39,9 @@ SponsorGroup.propTypes = {
 SponsorGroup.styles = StyleSheet.create({
   group: {display: 'flex'},
+  sectionDivider: {
+    borderTop: '1px solid #9A9A9A',
```
Where'd you get this color? Could we maybe move it to [here](https://github.com/javascriptair/site/blob/7f60730b3bcafb2fab232384054c593e57216113/src/styles/variables.js)?

kentcdodds commented on src/pages/deals/index.js:
```diff
@@ -12,7 +12,7 @@ function DealsPage({deals}) {
       title="JavaScript Air Deals"
     >
-      <div className={`container ${css(styles.dealsPageRoot)}`}>
+      <div className={css(styles.dealsPageRoot)}>
```
love seeing all of these `container` classes being removed!

kentcdodds commented:
This looks fantastic! Just commented on a few things. This is really coming together and looking much cleaner :+1:

Commit:
Added divider color to variables file

Commit:
Merge branch 'master' into remove-legacy-css

kentcdodds commented:
Looks great! Thanks so much!
